### ISSUE TITLE
TFlite conversation failed

### ISSUE THREAD

**Issue by smitshilu** [tensorflow/tensorflow#18612]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac 10.13.3
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**:  1.6
- **Python version**: 3.6.3
- **Bazel version (if compiling from source)**: 0.9
- **CUDA/cuDNN version**: 9/7
- **GPU model and memory**: 1080Ti & 12G
- **Exact command to reproduce**:

I wrote MobileNet code to train a model from scratch. Also tried using MobileNet keras function for training but none of the model converts to tflite.

I am running following command

```
./bazel-bin/tensorflow/contrib/lite/toco/toco --input_file=/MobileNet-trined.pb --input_format=TENSORFLOW_GRAPHDEF  --output_format=TFLITE --output_file=train-1.tflite --inference_type=FLOAT --input_arrays=input_1 --output_arrays=dense_1/Sigmoid --allow_custom_ops
```

It gives me following result

```
2018-04-17 15:25:11.070338: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1236] Converting unsupported operation: RandomUniform
2018-04-17 15:25:11.098993: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 865 operators, 1364 arrays (0 quantized)
2018-04-17 15:25:11.139921: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After Removing unused ops pass 1: 809 operators, 1280 arrays (0 quantized)
2018-04-17 15:25:11.183528: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 809 operators, 1280 arrays (0 quantized)
1984959600
2006980784
2018-04-17 15:25:11.184242: F tensorflow/contrib/lite/toco/graph_transformations/resolve_batch_normalization.cc:90] Check failed: mean_shape.dims() == multiplier_shape.dims() 
Abort trap: 6
```

**Comment by svarjo** [comment]
Duplicate of https://github.com/tensorflow/tensorflow/issues/17684
As GarryLau points out at issue 17684 - do not use training graph but export evaluation graph that you then freeze.

**Comment by smitshilu** [comment]
@svarjo Can you help me with how to generate eval.pbtxt? I tried using mobilenet_v1_eval but it generates nothing

**Comment by svarjo** [comment]
Study export_inference_graph.py example in slim folder for exporting inference graph. After that you have to combine model with the trained variables using tensorflow freeze_graph tool (ie call for example 'python3 -m tensorflow.python.tools.freeze_graph' with suitable input_graph and checkopoint).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 47 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 62 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by raghuraman-k** [comment]
Please note that you need to create an eval graph, freeze it and send it to toco for conversion. Here it looks like you are using a training graph. If you are using Keras, you can export to saved model and pass that to tflite_convert (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/python/tflite_convert.py)

**Comment by gargn** [comment]
Please reopen if you are still encountering this issue.
